src/python/geoclaw/topotools.py

```diff
@@ -773,28 +800,50 @@ def read_header(self):
                         values.append(v)
                     except:
                         pass
+                dx = values[0]
                 if len(values) == 1:
-                    self._delta = (values[0], values[0]) # if only dx given
+                    dy = dx   # only dx given
                 elif len(values) == 2:
+                    dy = values[1]
                     self._delta = (values[0], values[1])  # if dx,dy on line
                 else:
                     raise IOError("Cannot parse dx,dy line: %s" % line)
+                self._delta = (dx, dy)
                     
                         
                 self.no_data_value = float(topo_file.readline().split()[value_index])
+
+                x = numpy.linspace(xll, xll+(num_cells[0]-1)*dx, num_cells[0])
+                y = numpy.linspace(yll, yll+(num_cells[1]-1)*dy, num_cells[1])
+                if self.grid_registration in ['lower', 'llcenter']:
+                    # x,y are cell center / data locations:
+                    self._x = x
+                    self._y = y
+                elif self.grid_registration == 'llcorner':
+                    # x,y are lower left corners:
+                    # data points are offset by dx/2, dy/2
+                    self._x = x + dx/2.
+                    self._y = y + dy/2.
+                    print('*** Note: since grid registration is llcorner,')
+                    print('    will shift x,y values by (dx/2, dy/2) to cell centers')
+                else:
+                    # assume that x,y are cell center / data locations:
+                    self._x = x
+                    self._y = y
+                    print('*** Warning: Unrecognized grid_registration: %s' \
```

Comment:
Might want to instead raise a warning so it could be caught in principle.  See [this stackoverflow post](https://stackoverflow.com/questions/15933741/how-do-i-catch-a-numpy-warning-like-its-an-exception-not-just-for-testing#15934081) for an example.